flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/rules/logical/LogicalCorrelateToJoinFromTemporalTableFunctionRule.scala

```diff
@@ -155,6 +155,7 @@ class LogicalCorrelateToJoinFromTemporalTableFunctionRule
     case hep: HepRelVertex => getRelOptSchema(hep.getCurrentRel)
     case single: SingleRel => getRelOptSchema(single.getInput)
     case bi: BiRel => getRelOptSchema(bi.getLeft)
+    case union: Union => getRelOptSchema(union.getInput(0))
```

Comment:
I think the this fix still not completed, there are still some other RelNodes would cause NPE, the fix like following may be better, what do you think?
    case relNode: RelNode if relNode.getInputs.isEmpty => relNode.getTable.getRelOptSchema
    case _ => getRelOptSchema(relNode.getInput(0))

Reply:
Done